crates/hyperswitch_connectors/src/connectors/jpmorgan/transformers.rs

```diff
@@ -142,24 +375,191 @@ impl<F, T> TryFrom<ResponseRouterData<F, JpmorganPaymentsResponse, T, PaymentsRe
     }
 }
 
-//TODO: Fill the struct with respective fields
-// REFUND :
-// Type definition for RefundRequest
-#[derive(Default, Debug, Serialize)]
-pub struct JpmorganRefundRequest {
-    pub amount: StringMinorUnit,
+#[derive(Default, Debug, Deserialize, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct JpmorganCaptureRequest {
+    capture_method: Option<CapMethod>,
+    amount: MinorUnit,
+    currency: Option<common_enums::Currency>,
 }
 
-impl<F> TryFrom<&JpmorganRouterData<&RefundsRouterData<F>>> for JpmorganRefundRequest {
+#[derive(Debug, Default, Copy, Serialize, Deserialize, Clone)]
+#[serde(rename_all = "UPPERCASE")]
+pub enum CapMethod {
+    #[default]
+    Now,
+    Delayed,
+    Manual,
+}
+
+impl TryFrom<&JpmorganRouterData<&PaymentsCaptureRouterData>> for JpmorganCaptureRequest {
     type Error = error_stack::Report<errors::ConnectorError>;
-    fn try_from(item: &JpmorganRouterData<&RefundsRouterData<F>>) -> Result<Self, Self::Error> {
+    fn try_from(
+        item: &JpmorganRouterData<&PaymentsCaptureRouterData>,
+    ) -> Result<Self, Self::Error> {
+        let capture_method = Some(map_capture_method(
+            item.router_data.request.capture_method.unwrap_or_default(),
+        )?);
         Ok(Self {
-            amount: item.amount.to_owned(),
+            capture_method,
+            amount: item.amount,
+            currency: Some(item.router_data.request.currency),
         })
     }
 }
 
-// Type definition for Refund Response
+#[derive(Debug, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct JpmorganCaptureResponse {
+    pub transaction_id: String,
+    pub request_id: String,
+    pub transaction_state: JpmorganTransactionState,
+    pub response_status: JpmorganTransactionStatus,
+    pub response_code: String,
+    pub response_message: String,
+    pub payment_method_type: PaymentMethodTypeCapRes,
+}
+
+#[derive(Default, Debug, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct PaymentMethodTypeCapRes {
+    pub card: Option<CardCapRes>,
+}
+
+#[derive(Default, Debug, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct CardCapRes {
+    pub card_type: Option<Secret<String>>,
+    pub card_type_name: Option<Secret<String>>,
+    unmasked_account_number: Option<Secret<String>>,
+}
+
+impl From<JpmorganTransactionState> for common_enums::AttemptStatus {
+    fn from(item: JpmorganTransactionState) -> Self {
+        match item {
+            JpmorganTransactionState::Authorized => Self::Authorized,
+            JpmorganTransactionState::Closed => Self::Charged,
+            JpmorganTransactionState::Declined | JpmorganTransactionState::Error => Self::Failure,
+            JpmorganTransactionState::Pending => Self::Pending,
+            JpmorganTransactionState::Voided => Self::Voided,
+        }
+    }
+}
```

Comment:
Why is this function in code if it is not being used anywhere?

Reply:
removed